### DISCUSSION TITLE
Preset Window Sizes

### DISCUSSION THREAD

**Discussion by LOuroboros** [ajohns6/Zelda-3-Launcher#4] [Ideas]
Reposting what I mistakenly brought up as an Issue ticket in #3.

I'm not sure if you're accepting feature requests for the launcher at all, so I'll just take the plunge.

I just found out the existence of this launcher thanks to [this post on Reddit](https://www.reddit.com/r/emulation/comments/110m3ce/a_link_to_the_past_pc_port) and I immediately thought about one thing I'd personally like to see added to the launcher's configuration options; preset window sizes.

That is, the ability to make the launcher set the game's window size to windowed mode, and more specifically, set its dimensions based on a valid window size preset, such as the native internal resolution of the game on the SNES (256x224?), 2x that size (512x448), etc. Kind of like many emulators on PC do.

Thanks in advance for reading!

**Comment by RadzPrower** [comment]
First of all, I can't actually change the behavior of the game itself as all I am doing is managing existing settings. This project is wholely separate from the game development.

Secondly, it kind of sounds like the settings will allow for what you want to do now with window scaling and even custom window sizes.

Maybe you are wanting something even more specific than that, but if that's the case, you would have to take it up with snesrev, the actual zelda3 developer.

**Reply by LOuroboros** [reply]
> First of all, I can't actually change the behavior of the game itself as all I am doing is managing existing settings. This project is wholely separate from the game development.

And I suppose that the game's window's size/dimensions is not something that the launcher can mess with, is that it?

> Secondly, it kind of sounds like the settings will allow for what you want to do now with window scaling and even custom window sizes.

Cool!

> Maybe you are wanting something even more specific than that, but if that's the case, you would have to take it up with snesrev, the actual zelda3 developer.

I don't know where it should go, to be honest. Since your launcher already incorporated many tweaks and modifications to how the game's window is spawned, I figured this was the right place?

The only thing I suggested is an option in the launcher to preemptively choose the dimensions of the game's window, just like it can already preemptively set the Screen Aspect Ratio, that's all in case I wasn't clear enough 😅

**Reply by RadzPrower** [reply]
Something like this:
![image](https://user-images.githubusercontent.com/37708128/218574004-7dc91a4c-de5f-445c-acaa-e55390d259ad.png)

Or this:
![image](https://user-images.githubusercontent.com/37708128/218574071-68350c98-ac15-448b-bc50-e8fabb7f5fe0.png)

Or did you want a prompt to ask you how you wanted to launch it?

**Reply by LOuroboros** [reply]
> Something like this: ![image](https://user-images.githubusercontent.com/37708128/218574004-7dc91a4c-de5f-445c-acaa-e55390d259ad.png)
> 
> Or this: ![image](https://user-images.githubusercontent.com/37708128/218574071-68350c98-ac15-448b-bc50-e8fabb7f5fe0.png)

Yup! That's exactly it.

Window Scaling 2 would make the game's window spawn as a 512x448 window, yes? That's quite effectively the sort of thing I was suggesting to implement 😄

**Reply by RadzPrower** [reply]
The `Custom` window size overwrites the scaling since it's a window size rather than a resolution, but the scaling will be based on the default resolutions if you use the `Auto` window size. By default, the aspect ratio will always remain the same regardless of window size:
![image](https://user-images.githubusercontent.com/37708128/218577411-a7bc72aa-3f53-4f06-b44f-95865671d8b1.png)

As you can see, even at 1024 width, the aspect ratio remains the same 4:3.
